app/scripts/modules/core/src/function/FunctionDetails.tsx

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { Overridable, IOverridableProps } from 'core/overrideRegistry';
+
+export interface IFunctionDetailsProps extends IOverridableProps {}
+
+@Overridable('function.details')
+export class FunctionDetails extends React.Component<IFunctionDetailsProps> {
```

Comment:
We could also make this functional since it is presentational one-liner

Reply:
Unfortunately, it looks like our `@Overridable` annotation can only decorate class components.

https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/overrideRegistry/Overridable.tsx#L36-L40

At least in its current form. 😞

Reply:
Thanks for pointing this out. A `PureComponent` could be a viable alternative in the current form, I would have to look at the differences between the two interfaces in react.